addons/account_edi/models/account_edi_format.py

```diff
@@ -505,17 +510,16 @@ def _retrieve_product(self, name=None, default_code=None, barcode=None):
             # cut Sales Description from the name
             name = name.split('\n')[0]
         domains = []
-        if default_code:
-            domains.append([('default_code', '=', default_code)])
         if barcode:
             domains.append([('barcode', '=', barcode)])
-
-        # Search for the product with the exact name, then ilike the name
-        name_domains = [('name', '=', name)], [('name', 'ilike', name)] if name else []
-        for name_domain in name_domains:
+        if default_code:
+            domains.append([('default_code', '=', default_code)])
+        if name:
+            domains += [[('name', '=', name)], [('name', 'ilike', name)]]
+        for domain in domains:
```

Comment:
The search in a loop is not ideal...
would it be possible to use `OR` and then `filtered_domain`?